client/extensions/woocommerce/app/promotions/promotion-create.js

```diff
@@ -154,8 +159,10 @@ function mapStateToProps( state ) {
 	const currency = currencySettings ? currencySettings.value : null;
 	const promotionId = getCurrentlyEditingPromotionId( state, site.ID );
 	const promotion = promotionId ? getPromotionWithLocalEdits( state, promotionId, site.ID ) : null;
+	const hasEdits = Boolean( getPromotionEdits( state, promotionId, site.ID ) );
 
 	return {
+		hasEdits,
```

Comment:
Could add `hasEdits` to `propTypes` too.

Reply:
Good catch, thanks!